docs/integrations/airflow/operator.md

```diff
@@ -1,50 +1,189 @@
 ---
-sidebar_position: 2
+sidebar_position: 3
 ---
 
-# Lineage in custom operators
+# Exposing lineage in Airflow operators
 
-As a way of extending the OpenLineage Airflow integration, we are providing a way to [add custom extractors](./extractor.md). 
-This is great for people who want to add integration with external operators - for example, a regular Airflow user who wants to support, for example, `GCSToS3Operator`.
+Since: 0.18.0
 
-However, it's too cumbersome for people who own operators, and want to add the default implementation of OpenLineage for their operators for external users. 
-This is why we have an additional way to expose lineage: the `get_openlineage_facets_on_start` and `get_openlineage_facets_on_complete` methods that can be implemented on your Operator.<br />When any of those are present, OpenLineage's `DefaultExtractor` uses them to get lineage data. If you don't define `get_openlineage_facets_on_complete` it will run `get_openlineage_facets_on_start` instead.
+## Scope
+Lineage extraction logic should be as close as possible to the operator logic. (for airflow operators in the airflow repository, for other providers in their own repository). This makes lineage extraction more stable as it lives with the operators. 
 
-For example, let's assume you want to expose lineage from an operator that copies data from one table to another:
+Previously the OpenLineage library required one `Extractor` for each supported `Operator` which is brittle and can break when operator internals changes. 
+It's too cumbersome for people who own operators, and want to add the default implementation of OpenLineage for their operators for external users. 
+This is still an option when you can't modify the operator itself: See [add custom extractors](./extractors/custom-extractors.md). 
+
+Each operator is responsible for describing lineage per the spec below, but the actual lineage events are still being sent by the OpenLineage library in the TaskInstanceListener.
+
+## Context
+OpenLineage collects the following information regarding the Datasets being read and written by a task:
+
+ - Dataset name and namespace [required] - the format for naming is outlined in the [naming specification](https://github.com/OpenLineage/OpenLineage/blob/main/spec/Naming.md#datasets).
+ - Dataset schema [optional] - The column names and types, if known. Complex types, like structs and arrays are supported
+ - Query id [optional] - for systems that expose an identifier, the id of the query (a Run facet, not a Dataset facet, but is always exposed by the Data Source’s proprietary API). For example operators for Bigquery, Redshift, and Snowflake should all allow this.
```

Comment:
- Query id [optional] - for systems that expose an identifier, the ID of the query. This is a Run facet, not a Dataset facet, but it is often exposed by the Data Source’s proprietary API. For example, operators for Bigquery, Redshift, and Snowflake should all allow this.